src/app.rs

```diff
@@ -16,6 +17,7 @@ pub struct App {
 
 #[derive(Debug)]
 pub struct UiState {
+    pub end_date: DateTime<Utc>,
```

Comment:
Hmm... Maybe we can use `start_date` instead? WDYT? I'm not sure which makes more sense. This probably requires some domain knowledge... I'll try to read up.

Reply:
Actually, maybe we should have both, no? We might want to support custom start / end date in the future. Not sure... Haha.

Reply:
Reference: https://support.stockcharts.com/doku.php?id=sharpcharts:indicator_past_data

Reply:
Yea we can do that.